### PR TITLE
Corrected casting problem for singular/plural values and included UnitTests for the casting cases

### PR DESCRIPTION
**Pull request by abdermendz**
#### Description: 

Pull request containing fix for issue: LMCROSSITXSADEPLOY-1700

#### Issue: 
Problem with casting arguments and lack of appropriate message.

### CODE REVIEW

**Review by radito3 — CHANGES_REQUESTED** [review]
Import our eclipse_formatter.xml into your IDE and fix the formatting of the new code + revert the formatting changes to old code

File: multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/util/PropertiesUtil.java [hunk 1]
```diff
@@ -120,15 +114,17 @@ private static boolean canGetParameter(Map<String, Object> parameters, String pa
...
-        if (values != null) {
-            result.addAll(values);
+        Object pluralValues = getPropertyValue(propertiesList, plural, null);
+        if (pluralValues == null) {
+            Object singleValue = getPropertyValue(propertiesList, single, null);
+            if (singleValue == null) {
+                return result;
+            }
+            result.add((T) singleValue);
             return result;
```
Comment (radoslav-d) [inline multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/util/PropertiesUtil.java:124]: Could be:
```
if (singleValue != null) {
    result.add((T) singleValue);
}
return result;
```

File: multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/PlaceholderResolver.java [hunk 2]
```diff
@@ -42,10 +43,17 @@ protected Object getSingularParameter(String plural, List<Map<String, Object>> p
...
+                validateList(parameters.get(plural), plural);
                 pluralValue = (List<Object>) parameters.get(plural);
             }
         }
         return pluralValue == null ? null : pluralValue.get(0);
     }
 
+    private void validateList(Object toValidate, String plural) {
+        if (!(toValidate instanceof List)) {
+            throw new ContentException("Invalid type provided for " + plural
```
Comment (radito3) [inline multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/PlaceholderResolver.java:55]: Now that the message is extracted in a constant, use it

File: multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/util/PropertiesUtil.java [hunk 3]
```diff
@@ -117,17 +117,18 @@ private static boolean canGetParameter(Map<String, Object> parameters, String pa
...
+            result.addAll((List<T>) pluralValues);
             return result;
         }
-        T value = (T) getPropertyValue(propertiesList, single, null);
-        if (value != null) {
-            result.add(value);
+        Object singleValue = getPropertyValue(propertiesList, single, null);
+        if (singleValue != null) {
+            result.add((T) singleValue);
+            return result;
```
Comment (radito3) [inline multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/util/PropertiesUtil.java:131]: No need for this line

**Pull request merged**